File: server/java/src/main/java/com/onlinepayments/example/api/models/paymentlinks/CreatePaymentLinkRequest.java
package com.onlinepayments.example.api.models.paymentlinks;

import com.onlinepayments.example.service.domain.common.enums.Currency;
import com.onlinepayments.example.service.domain.paymentlinks.Action;
import com.onlinepayments.example.service.domain.paymentlinks.ValidityPeriod;

import java.math.BigDecimal;

public record CreatePaymentLinkRequest(
        BigDecimal amount,
        Currency currency,
        String description,
        ValidityPeriod validFor,
        Action action
) { }

File: server/dotnet/sdk-dotnet-example/Presentation/Models/CreatePaymentLinks/CreatePaymentLinkRequest.cs
using Business.Domain.Common.Enums;
using Business.Domain.Enums.PaymentLinks;
using Action = Business.Domain.Enums.PaymentLinks.Action;

namespace Presentation.Models.CreatePaymentLinks;

public class CreatePaymentLinkRequest
{
    public decimal? Amount { get; init; }
    
    public Currency? Currency { get; init; }

    public string? Description { get; init; }

    public ValidityPeriod? ValidFor { get; init; }

    public Action? Action { get; init; }
}